nvflare/apis/server_engine_spec.py

```diff
@@ -115,3 +115,15 @@ def persist_components(self, fl_ctx: FLContext, completed: bool):
 
     def restore_components(self, snapshot: RunSnapshot, fl_ctx: FLContext):
         pass
+
+    def send_admin_requests(self, requests):
+        pass
+
+    def check_client_resources(self, resource_reqs):
+        pass
+
+    def cancel_client_resources(self, resource_check_results, resource_reqs):
+        pass
+
+    def start_client_job(self, client_sites, run_number):
+        pass
```

Comment:
We might need to add docstrings to these methods to explain why they need to be here